buildscripts/docker-entrypoint.sh

```diff
@@ -22,6 +22,69 @@ if [ "${1}" != "minio" ]; then
     fi
 fi
 
+# helper function to check if a string contains another string
+string_match () { 
+    case "$2" in 
+        $1) return 0 ;; 
+        *) return 1 ;;
+    esac ; 
+}
+
+# switch to non root (minio) user if there isn't an existing Minio data directory or
+# if its an empty data directory. We dont switch to minio user if there is an existing
+# data directory, this is because we'll need to chown all the files -- which may take 
+# too long, ruining the user's experience.
+docker_switch_non_root() {
+    arguments_processed=0
+    # loop through all the arguements
+    for var in "$@"
+    do  
+        # ignore minio, server, gateway or minio server flags
+        if [ "${var}" = "minio" ] || [ "${var}" = "server" ] \
+            || [ "${var}" = "gateway" ] || string_match '--*' "${var}" || \
+            string_match ':*' "${var}" ; then
+            # count as processed
+            arguments_processed=$((arguments_processed+1))
+            continue
+        fi
+        # strip scheme and hostname if present in an arg
+        if [ "${var:0:4}" = "http" ]; then
+            var=$(printf -- "%s"  "${var}" | cut -d/ -f4-)
+        fi
+        # if the given directory exists
+        if [ -d "${var}" ];then
+            echo "directory $var exists"
+            # if the directory is empty
+            if [ "$(ls -A $var)" = "" ]; then
+                # change owner to minio user
+                chown -R minio:minio "${var}"
+            fi 
+            # find the owner of the directory
+            owner=$(ls -ld "${var}" | awk '{print $3}')
+            # if minio is the owner, count as processed
+            if [ "${owner}" = "minio" ];then
+                echo "previous owner = minio"
```

Comment:
All messages should be capitalized when printed and also should be a bit more descriptive of what they did rather than looking like a "return" statement.

Reply:
This was added for debugging, I inadvertently committed. I don't intend to add echo messages, do you think we should have messages for the user here?